### ISSUE TITLE
Usage with Chatterino?

### ISSUE THREAD

**Issue by Icarus77** [streamlink/streamlink-twitch-gui#607]
### Checklist

- [X] I understand the difference between *Streamlink* and *Streamlink Twitch GUI*.
- [X] I tried solving my issue by reading the project's wiki first.
- [X] This is not a bug report or feature request and I have read the contribution guides.

I changed from Chatty to Chatterino and it seems to not be an option for Streamlink.Twitch GUI. When I change to a custom application, Chatterino only opens, but not the chat at all, and it opens in a new window if I already have one open instead of a new tab.

Can Chatterino not function properly with this GUI?

**Comment by bastimeyer** [comment]
You need to specify the launch parameters for custom applications. Otherwise, how should the custom application know what you're trying to do?
https://github.com/streamlink/streamlink-twitch-gui/wiki/Chat-configuration#custom-application

And the chosen application of course also needs to support it, but it doesn't look like Chatterino does (yet):
https://github.com/fourtf/chatterino2/issues/43

**Comment by zneix** [comment]
Just wanted to mention that an option to use CLI arguments for certain channels to open on startup was recently added to Chatterino (chatterino/chatterino2#1940)

**Comment by Maquata** [comment]
@zneix Could you explain how to use these CLI arguments with Streamlink Twitch GUI? I've tried putting "--channels {channel}" and similar variations in the application parameters under settings with no success.

**Comment by CristenPerret** [comment]
I also am having an issue setting this feature up with Streamlink's GUI.
All I want it to do is when I press the 'Chat' Button on the GUI it will populate that in aChattarino new tab.
Not a static parameter of a specific stream or user, just the one im pushing via the streamlink app.
Idc even if it doesnt check for dupes, ill just [x] it out if thats the case. What param would this be?

**Comment by bastimeyer** [comment]
What Chatterino has implemented doesn't seem to work.

```
chatterino -c shroud
chatterino -c t:shroud
chatterino --channels shroud
chatterino --channels t:shroud
```

This means you cannot use it from Streamlink Twitch GUI. It also doesn't make much sense, since they don't seem to have a feature for turning the application into a singleton app, and opening a new instance for each stream is an anti-feature when the application itself supports tabs and split views.

Btw, [their implementation of the linked PR that got merged uses unsanitized inputs while parsing a JSON object](https://github.com/Chatterino/chatterino2/blob/ec94869480ab94e8852bdaa2b2c4e286f2969942/src/common/Args.cpp#L78). That's not how you do it. Tell them to re-implement and fix their stuff, otherwise a Catterino profile can't be added here.

**Comment by CristenPerret** [comment]
The GUI works seamlessly with Chatty though. O well, that'd be a really convenient feature and I cant imagine any malicious use from like bots or whatever with it

**Comment by leon-richardt** [comment]
> What Chatterino has implemented doesn't seem to work.

Just wanted to give an update: Since https://github.com/Chatterino/chatterino2/pull/2368 got merged yesterday, the feature now works as originally intended again. It is currently only available on nightly builds but will be included in the next stable/beta release.

**Comment by Sukid** [comment]
Yep, it works now!! Although making --channels {channel} the argument parameter for some reason makes Chatterino join a channel with a bunch of /s inserted into the name, like /f/l/u/t/t/e/n/ instead of flutten for example.

**Comment by bastimeyer** [comment]
A Chatterino chat preset has been added in 127a7a9 with default executable names and paths for Windows, macOS and Linux.

If a singleton CLI argument similar to Chatty's `-single` gets added to Chatterino, please ping me, so it can be added here and the preset improved.

Since there are no other useful CLI arguments available at the moment, the preset doesn't have a field for custom args.

**Comment by leon-richardt** [comment]
> If a singleton CLI argument similar to Chatty's `-single` gets added to Chatterino, please ping me, so it can be added here and the preset improved.

Will do

**Comment by mrk-han** [comment]
I'm looking for a way to open channels in Streamlink-Twitch-Gui from Chatterino. When I was going the other way, from GUI to Chatterino -- Chatterino would open a new window with only one chat channel open (seemed like a separate instance completely). Also, it's just easier for me to use Chatterino as the main way of opening streams since I always have it up.


**Desired steps would be:**
1) Open Chatterino (All channels are saved by default)
2) Right click on channel header in chatterino
3) Open in Custom Player --- and set Streamlink-Twitch-Gui as the custom player

Is this possible? I tried fiddling around with it for about an hour now with no luck. 

This gives me the benefit of using Chatterino as the main repository of my channels I like to swap between with using the incredible benefits of streamlink-twitch-gui to control all of the settings etc of streamlink opening MPV --- which thanks to y'all I've gotten <1 latency.

Sorry to post on this closed comment, and thanks for the time.

**Comment by bastimeyer** [comment]
> I'm looking for a way to open channels in Streamlink-Twitch-Gui from Chatterino.
> Is this possible?

https://github.com/streamlink/streamlink-twitch-gui/wiki/Parameters#--launch

The `--launch` parameter is currently broken on `2.0.0` due to the Twitch API rewrite and missing tests for this parameter, so it got unnoticed, which means you'll have to build from the master branch or wait for `2.1.0` which I'm going to publish soon(ish):
d5d519c401104e745b7f952b8df01dba23adf8bf

https://github.com/streamlink/streamlink-twitch-gui/blob/master/CONTRIBUTING.md#developing-and-building

**Comment by mrk-han** [comment]
@bastimeyer Thank you so much! 

I think I'll have to wait for the 7tv V3 Release and its support/integration to be added into the main Chatterino branch. I'm using the Chatterino7 fork right now.

**Comment by TubaApollo** [comment]
Although I've read this thread a few times now, I don't really understand if there was a solution to the multiple instances problem. And if so, how should it be done properly? It's a bit irritating for me, but it would be great to open new tabs instead of having multiple instances of Chatterino.

**Comment by mrk-han** [comment]
@TubaApollo Sounds like if that flag is in then it should work (assuming 2.1.0 was released) -- I haven't checked what the latest release is though.

**Comment by bastimeyer** [comment]
> it would be great to open new tabs instead of having multiple instances of Chatterino.

@TubaApollo As you can read in my previous comment, there is no option for that in Streamlink Twitch GUI, because Chatterino didn't provide a CLI argument for such a feature when the preset was added. According to the linked threads you can see above, there still isn't one:
https://github.com/Chatterino/chatterino2/issues/2498

Chatty for example does have the `-single` flag, which turns it into a singleton process, so tabs can be opened instead.

And implementing any other kind of inter-process-communication (if there even was an IPC mechanism supported by Chatterino) probably won't be done here, as it'd require re-implementing the entire `ChatService`, and I'm not willing to spend (waste) my time on that.

> Sounds like if that flag is in then it should work (assuming 2.1.0 was released) -- I haven't checked what the latest release is though.

@mrk-han You are talking about something else, launching Streamlink Twitch GUI from Chatterino. This was never the topic of this thread, same as your previous comment. Streamlink Twitch GUI's `--launch` parameter has been fixed in 2.1.0 which was released in May. 2.2.0 has been released yesterday.
https://github.com/streamlink/streamlink-twitch-gui/blob/master/CHANGELOG.md

**Comment by TubaApollo** [comment]
Thank you so much for clarifying, i spend so much time trying to get it working since i thought it works somehow work. I was just confused.
